### PR TITLE
[SofaGeneralEngine] add test and minor cleaning for IndexValueMapper

### PR DESCRIPTION
**Pull request by ErwanDouaille**
Introduction (by Damien)
Erwan is a new member of Defrost. He will work on sofa, cleaning the haptic plugins & finally make user interfaces. Here is his first PR (to practice writing tests and our workflow :))

This PR contains tests to validate the behavior of the IndexValueMapper engine which was not tested before. 
The component was selected as it is listed into the SofaNGExt list. 

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by damienmarchal** [comment]
Let's build it [ci-build]

**Comment by damienmarchal** [comment]
@ErwanDouaille I think the CMakeList.txt is missing in the PR.

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by guparan** [comment]
Hi @ErwanDouaille,
Thank you for your first PR in SOFA and welcome in the community 😄 
[ci-build]

File: modules/SofaGeneralEngine/IndexValueMapper.h [hunk 1]
```diff
@@ -73,15 +71,15 @@ class IndexValueMapper : public sofa::core::DataEngine
...
+    Data<sofa::helper::vector<Index> > f_indices; ///< Indices to map value on
+    Data<Real> f_value; ///< Value to map indices on
 
     //Output
-    Data<sofa::helper::vector<Real> > f_outputValues;
+    Data<sofa::helper::vector<Real> > f_outputValues; ///< New map between indices and values
 
     //Parameter
-    Data<Real> p_defaultValue;
+    Data<Real> p_defaultValue; ///< Default value for indices without any value
```
Comment (guparan) [inline modules/SofaGeneralEngine/IndexValueMapper.h:82]: Nice Doxygen inline comments ❤️

Comment (ErwanDouaille) [reply modules/SofaGeneralEngine/IndexValueMapper.h:82]: Thanks !

I just respect the rules :D

**Pull request merged**